src/osdatahub/NamesAPI/names_api.py

```diff
@@ -0,0 +1,160 @@
+                (EPSG:27700) CRS
+            local_type (Union[Iterable, str], optional): Filters the results to certain local types. Available local
+                types can be found at the bottom of https://osdatahub.os.uk/docs/names/technicalSpecification
+
+        Returns:
+            FeatureCollection: The results of the query in GeoJSON format
+        """
+        data = GrowList()
+        params = {"query": text}
+
+        if limit <= 0:
+            raise ValueError("Parameter \"limit\" must be a positive integer")
+
+        if bounds:
+            if not bounds.crs == "EPSG:27700":
+                raise TypeError("Bounds must be in British National Grid CRS (EPSG:27700)")
+            params.update({"bounds": bounds.bbox.to_string()})
+        if bbox_filter or local_type:
+            if bbox_filter and bbox_filter.crs != "EPSG:27700":
+                raise TypeError("Bounding Box filter must be in British National Grid CRS (EPSG:27700)")
+            params.update({"fq": self.__format_fq(bbox_filter, local_type)})
+
+        try:
+            n_required = min(limit, 100)
+            while n_required > 0 and data.grown:
+                params.update({"offset": len(data), "maxresults": n_required})
+                print(params)
+                response = requests.get(self.__endpoint("find"), params=params)
+                data.extend(self.__format_response(response))
+                n_required = min(100, limit - len(data))
+        except KeyError:
+            response.raise_for_status()
+        return addresses_to_geojson(data.values, "EPSG:27700")
+
+    def nearest(self, point: tuple, radius: float = 100, local_type: Union[Iterable, str] = None) -> FeatureCollection:
+        """Takes a pair of coordinates (X, Y)/(Lon, Lat) as an input
+        to determine the closest name.
+
+        Args:
+            point (tuple): A set of coordinates in British National Grid (EPSG:27700) format
+            radius (float): The search radius in metres (max. 1000).
+                Defaults to 100.
+            local_type (Union[Iterable, str], optional):  Filters the results to certain local types. Available local
+                types can be found at the bottom of https://osdatahub.os.uk/docs/names/technicalSpecification
+
+        Returns:
+            FeatureCollection: The results of the query in GeoJSON format
+        """
+        data = GrowList()
+        if not all([str(p).isnumeric() for p in point]):
+            raise TypeError("All values in argument \"point\" must be numeric")
+        if not 0.01 <= radius <= 1000:
+            raise ValueError(f"Argument \"radius\" must be between 0 and 1000, but had value {radius}")
+
+        params = {"point": ",".join([str(c) for c in point]), "radius": radius}
+        if local_type:
+            params.update({"fq": self.__format_fq(local_type=local_type)})
+        try:
+            response = requests.get(self.__endpoint("nearest"), params=params)
```

Comment:
Is it certain that the "nearest" api never requires pagination?

Reply:
What do you mean by Pagination?

Reply:
The features API will chunk up responses with 100 features, does something like that happen here?

Reply:
Ahhh! Yes, the nearest endpoint only returns the nearest single address. Pagination never happens